pkg/services/ngalert/schedule/schedule.go

```diff
@@ -398,16 +401,35 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 
 		if err != nil || results.HasErrors() {
 			evalTotalFailures.Inc()
+
+			// If we have reached the maximum number of retries - no-op on these return operations.
+			if !noRetry {
```

Comment:
Nit: I would change the logic and use `retry` instead of `noRetry`, so you don't have to think `not noRetry` but just `retry`.